### PR TITLE
Add net/http client instrumentor

### PR DESCRIPTION
**Pull request by oxeye-gal**
Hi,

This PR attempts to add instrumentation for net/http client.

We do it by writing a fake tophash value (as calculating the correct one will be harder to achieve) to the first headers map bucket and writing the "traceparent" header key and value into memory.

This was tested with Go versions 1.12-1.19.

Our test environment consisted of instrumented Python application that sends a request to instrumented Go http server which in turn sends an http request to an additional instrumented Go http server.

The produced trace looks like the following:

213374473-d266bf1f-49f5-4f8c-8e0d-53ed51d80201

We would love to get your feedback on this.
Thanks!

### CODE REVIEW

**Review by damemi — CHANGES_REQUESTED** [review]
@oxeye-gal looks like this updates the test fixtures (which is expected since this is a new instrumentation) based on the [CI failure](https://github.com/open-telemetry/opentelemetry-go-instrumentation/actions/runs/4783367077/jobs/8520701861?pr=91#step:17:45)

If you have [kind](https://kind.sigs.k8s.io/) installed, you should be able to run `make fixture-nethttp` to re-generate the test output and commit the result.

**Comment by MikeGoldsmith** [comment]
Hey @oxeye-gal - this PR needs rebasing after we merged ARM support on #82. I’m happy to help do that if you would like us to.

**Comment by MikeGoldsmith** [comment]
@oxeye-gal now #92 has been merged, I’ll look to update this PR too 👍🏻

**Comment by MikeGoldsmith** [comment]
Picking this up now - sorry it's taken longer than expected to get back to.

**Comment by MikeGoldsmith** [comment]
@oxeye-gal please could you "update branch" one more time, and we can merge this PR :)

**Review by MikeGoldsmith — APPROVED** [review]

**Comment by MikeGoldsmith** [comment]
@open-telemetry/go-instrumentation-approvers Please can you review this PR - it needs a final "update branch" but once done I'd like to merge this. I've worked up making sure it works and plan on expanding the same work to the other HTTP probes once it's done (see #101).

File: pkg/instrumentors/bpf/net/http/client/bpf/probe.bpf.c
Comment (pdelewski) [inline pkg/instrumentors/bpf/net/http/client/bpf/probe.bpf.c:79]: Can we use constants instead of number literals?

**Review by pdelewski — APPROVED** [review]

**Review by edeNFed — APPROVED** [review]
lgtm

**Pull request merged**
